sdk/python/kfp/_client.py

```diff
@@ -141,12 +141,12 @@ def _load_config(self, host, client_id, namespace):
 
   def _is_iap_host(self, host, client_id):
     if host and client_id:
-      return re.match(r'\S+.endpoints.\S+.cloud.goog', host)
+      return re.match(r'\S+.endpoints.\S+.cloud.goog/pipeline', host)
     return False
 
   def _is_inverse_proxy_host(self, host):
     if host:
-      return re.match(r'\S+dot-datalab-vm\S+.googleusercontent.com', host)
+      return re.match(r'\S+dot-datalab-vm\S+.googleusercontent.com/{0,1}$', host)
```

Comment:
JFYI: `/{0,1}` is equivalent to `/?`

Reply:
I think it can also match the case where there is no slash?

Reply:
>I think it can also match the case where there is no slash?
Yes. Both of them. "Zero or one slash" vs. "optional slash".